test/test_other.py

```diff
@@ -2729,12 +2729,22 @@ def test_embind_closure_no_dynamic_execution(self):
 
   @is_slow_test
   @parameterized({
-    '': [[]],
-    'no_utf8': [['-sEMBIND_STD_STRING_IS_UTF8=0']],
-    'no_dynamic': [['-sDYNAMIC_EXECUTION=0']],
+    '': [{}],
+    'asserts': [{'ASSERTIONS': '1'}],
+    'no_utf8': [{'EMBIND_STD_STRING_IS_UTF8': '0'}],
+    'no_dynamic': [{'DYNAMIC_EXECUTION': '0'}],
   })
   @with_env_modify({'EMCC_CLOSURE_ARGS': '--externs ' + shlex.quote(test_file('embind/underscore-externs.js'))})
-  def test_embind(self, extra_args):
+  def test_embind(self, override_args_dict):
+    # Start with defaults and apply the test-specific overrides.
+    args_dict = {
+      'ASSERTIONS': '1',
+      'EMBIND_STD_STRING_IS_UTF8': '0',
+      'DYNAMIC_EXECUTION': '0'
+    }
+    for key, val in override_args_dict.items():
+      args_dict[key] = val
+    extra_args = [f'-s{key}={val}' for key, val in args_dict.items()]
```

Comment:
This seems a little over complicated.

If you want something this general we already have `-sRETAIN_COMPILER_SETTINGS` which basically does this but for all settings.   Maybe just use that?

Reply:
Unfortunately that option doesn't just work. It places the settings in a `compilerSettings` object inside the emcc output, but this test wraps that output in a function. Then when we try to read the setting it isn't in scope.

Reply:
For reference, here is the output if you want to see what I mean, and also the diff I used to get it: https://gist.github.com/kripken/bb18ae751e80771c34f56d53b42929ae

Reply:
If it doesn't work we should try to fix it.   Doesn't `-sEXPORTED_RUNTIME_METHODS=getCompilerSetting` work?

Reply:
Then I guess you would `cm.getCompilerSetting('DYNAMIC_EXECUTION')` (I'm not sure what cm is here but I guess its maybe "compiled module")?

Reply:
Ah, good point, the scoping issue is probably just the usual issue with needing to export. I'll try that.